osu.Game.Modes.Taiko/Scoring/TaikoScoreProcessor.cs

```diff
@@ -0,0 +1,270 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Game.Beatmaps;
+using osu.Game.Database;
+using osu.Game.Modes.Objects.Drawables;
+using osu.Game.Modes.Scoring;
+using osu.Game.Modes.Taiko.Judgements;
+using osu.Game.Modes.Taiko.Objects;
+using osu.Game.Modes.UI;
+using OpenTK;
+
+namespace osu.Game.Modes.Taiko.Scoring
+{
+    internal class TaikoScoreProcessor : ScoreProcessor<TaikoHitObject, TaikoJudgement>
+    {
+        /// <summary>
+        /// The maximum score achievable.
+        /// Does _not_ include bonus score - for bonus score see <see cref="bonusScore"/>.
+        /// </summary>
+        private const int max_score = 1000000;
+
+        /// <summary>
+        /// The amount of the score attributed to combo.
+        /// </summary>
+        private const double combo_portion_max = max_score * 0.2;
+
+        /// <summary>
+        /// The amount of the score attributed to accuracy.
+        /// </summary>
+        private const double accuracy_portion_max = max_score * 0.8;
+
+        /// <summary>
+        /// The factor used to determine relevance of combos.
+        /// </summary>
+        private const double combo_base = 4;
+
+        /// <summary>
+        /// The HP awarded by a <see cref="TaikoHitResult.Great"/> hit.
+        /// </summary>
+        private const double hp_hit_great = 0.03;
+
+        /// <summary>
+        /// The HP awarded for a <see cref="TaikoHitResult.Good"/> hit at HP >= 5.
+        /// </summary>
+        private const double hp_hit_good = 0.011;
+
+        /// <summary>
+        /// The HP awarded for a <see cref="TaikoHitResult.Good"/> hit at HP = 0.
+        /// <para>
+        /// Yes, this is incorrect, and goods at HP = 0 will award more HP than greats.
+        /// This is legacy and should be fixed, but is kept as is for now for compatibility.
+        /// </para>
+        /// </summary>
+        private const double hp_hit_good_max = hp_hit_good * 8;

```

Comment:
okay i see these are just base values which are interpolated later. you need to explain what these variables are better (nowhere is the `_max` explained).

also i'd just fix this rather than keeping broken "compatibility"